### ISSUE TITLE
type 'GeneratedMessage' can't be returned from method 'copyWith'

### ISSUE THREAD

**Issue by zaher-recode** [google/protobuf.dart#390]
so after upgrading to flutter 1.20 now the generated files *.pb.dart have errors when `super.copyWith` is generated.

example:
`Report copyWith(void Function(Report) updates) => super.copyWith((message) => updates(message as Report));`

the error says:
```A value of type 'GeneratedMessage' can't be returned from method 'copyWith' because it has a return type of 'Report' .dart(return_of_invalid_type) ```

I'm on  `libprotoc 3.12.4`

below is my flutter doctor -v if required:
```[✓] Flutter (Channel stable, 1.20.0, on Mac OS X 10.15.1 19B88, locale en)
    • Flutter version 1.20.0 at /Users/zaher/flutter
    • Framework revision 840c9205b3 (31 hours ago), 2020-08-04 20:55:12 -0700
    • Engine revision c8e3b94853
    • Dart version 2.9.0

 
[✓] Android toolchain - develop for Android devices (Android SDK version 29.0.3)
    • Android SDK at /Users/zaher/Library/Android/sdk
    • Platform android-29, build-tools 29.0.3
    • ANDROID_HOME = /Users/zaher/Library/Android/sdk
    • Java binary at: /Applications/Android
      Studio.app/Contents/jre/jdk/Contents/Home/bin/java
    • Java version OpenJDK Runtime Environment (build
      1.8.0_242-release-1644-b3-6222593)
    • All Android licenses accepted.

[✓] Xcode - develop for iOS and macOS (Xcode 11.3.1)
    • Xcode at /Applications/Xcode.app/Contents/Developer
    • Xcode 11.3.1, Build version 11C504
    • CocoaPods version 1.8.4

[✓] Android Studio (version 4.0)
    • Android Studio at /Applications/Android Studio.app/Contents
    • Flutter plugin version 46.0.2
    • Dart plugin version 193.7361
    • Java version OpenJDK Runtime Environment (build
      1.8.0_242-release-1644-b3-6222593)

[✓] VS Code (version 1.47.3)
    • VS Code at /Applications/Visual Studio Code.app/Contents
    • Flutter extension version 3.13.2
```

**Comment by Ahmed-gubara** [comment]
do you have `implicit-casts: false` in `analysis_options.yaml` ?
as a workaround, removing that line worked for me.

this #317 pull request seems like could solve this issue by using generic strongly typed `copyWith()` alternative

**Comment by zaher-recode** [comment]
Thanks,
Yeah Removing  `implicit-casts: false ` did solve the issue as  a workaround. 
Hope it can be fixed though

**Comment by tvh** [comment]
I managed to "fix" this by introducing more explicit casts. I did this the ugly way, but I assume it should be be easy enough by changing the code generation to do this instead. I just couldn't figure out how to make all the package versions line up.

```
find "lib/protos" -type f -exec sed -r \
    -e 's/as ([^ \)]+)\)\)/\0 as \1/g' \
    -i  \{\} \;
```

This will change
```
Settings copyWith(void Function(Settings) updates) => super.copyWith((message) => updates(message as Settings));
```
into
```
Settings copyWith(void Function(Settings) updates) => super.copyWith((message) => updates(message as Settings)) as Settings;
```

**Comment by tvh** [comment]
This seems to be fixed in 20.0.0.

**Comment by osa1** [comment]
Fixed with 780bee8c.
